pallets/order-book/src/lib.rs

```diff
@@ -420,6 +466,76 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::Weights::create_order())]
+		pub fn add_trading_pair(
+			origin: OriginFor<T>,
+			asset_in: T::AssetCurrencyId,
+			asset_out: T::AssetCurrencyId,
+			min_order: T::Balance,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			// We do not check, we just overwrite as this is an admin action.
+			TradingPairInOut::<T>::insert(asset_in.clone(), asset_out.clone(), min_order);
+			TradingPairOutIn::<T>::insert(asset_out.clone(), asset_in.clone(), ());
+
+			Self::deposit_event(Event::<T>::TradingPairAdded {
+				asset_in,
+				asset_out,
+				min_order,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::Weights::create_order())]
+		pub fn rm_trading_pair(
+			origin: OriginFor<T>,
+			asset_in: T::AssetCurrencyId,
+			asset_out: T::AssetCurrencyId,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			// We do not check, we just remove as this is an admin action.
+			TradingPairInOut::<T>::remove(asset_in.clone(), asset_out.clone());
+			TradingPairOutIn::<T>::remove(asset_out.clone(), asset_in.clone());
+
+			Self::deposit_event(Event::<T>::TradingPairRemoved {
+				asset_in,
+				asset_out,
+			});
+
+			Ok(())
+		}
+
+		/// Sets the minimum order amount for a given trading pair.
+		/// If the trading pair is not yet added this errors out.
+		#[pallet::call_index(6)]
+		#[pallet::weight(T::Weights::create_order())]
+		pub fn update_min_order(
+			origin: OriginFor<T>,
+			asset_in: T::AssetCurrencyId,
+			asset_out: T::AssetCurrencyId,
+			min_order: T::Balance,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			// try_mutate is a pain with the direct error return. But we do want to only
+			// update if the pair exists.
+			let _old_min_order = TradingPairInOut::<T>::get(&asset_in, &asset_out)?;
```

Comment:
Nit: Can't we use `try_mutate` for that?

Reply:
See comment above